### PR TITLE
Add regression checklist

### PR DESCRIPTION
**Pull request by brbrr**
This PR adds some initial setup for the regression checklist. Also, it adds a mapping.json that would be used for [test-case-reminder](https://github.com/apps/test-case-reminder) bot

#### Jetpack product discussion

https://wp.me/p9dueE-1wX

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Unfortunately, it is not possible to test the bot integration because it looks for configuration in master, not in a PR branch.
* Look through test-suites. Are they look good?

Once this merged, for new PRs that touches files in `mapping.json` test-case reminder bot will post a comment that will look like this:

image  

#### Proposed changelog entry for your changes:

* n/a

### CODE REVIEW

File: docs/testing/regression-checklist/regression-checklist.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+- Subdirectory
+- Single site
+- Subdirectory
+- Subdomain
+
+### WP versions
+
+- Current version
+- Current -1
+- With gutenberg plugin
```
Comment (bisko) [inline docs/testing/regression-checklist/regression-checklist.md:19]: Typo `Gutenberg` with a capital letteer.

File: docs/testing/regression-checklist/regression-checklist.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+- [ ] Site changes registered in Calypso’s Activity Log
+
+## Features
+
+- [ ] Publicize: Connect and share a post
+- [ ] SSO login
+- [ ] Stats registers views
+- [ ] Stats registers views in AMP views
+- [ ] Lazy loading
+- [ ] Site accelerator (make sure that images and core files are served from wpcom)
```
Comment (bisko) [inline docs/testing/regression-checklist/regression-checklist.md:58]: wpcom -> WPCOM or WordPress.com

File: docs/testing/regression-checklist/test-suites/extenstions/blocks/contact-form.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#### Contact Form
+
+- [ ] Publish a post with default form
```
Comment (bisko) [inline docs/testing/regression-checklist/test-suites/extenstions/blocks/contact-form.md:3]: Shouldn't this be Submit the form or something along the lines? Can it actually create a user-facing post?

File: docs/testing/regression-checklist/test-suites/modules/publicize.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#### Publicize
+
+- [ ] Publish a post with twitter integration
+- [ ] Publish a post with facebook integration
```
Comment (bisko) [inline docs/testing/regression-checklist/test-suites/modules/publicize.md:4]: Can we capitalize Facebook and Twitter?

File: docs/testing/regression-checklist/test-suites/extenstions/blocks/contact-form.md (hunk 3 above)
Comment (brbrr) [reply docs/testing/regression-checklist/test-suites/extenstions/blocks/contact-form.md:3]: oh, it's about publishing a post with Contact Form block, not about submitting the form. But this is indeed good test case :)

**Pull request merged**